### ISSUE TITLE
Typings don't work for redux@4.0.0

### ISSUE THREAD

**Issue by septs** [xdave/typescript-fsa-redux-thunk#11]
```
node_modules/typescript-fsa-redux-thunk/lib/index.d.ts:12:75 - error TS2314: Generic type 'Dispatch<A, S>' requires 2 type argument(s).

12 export declare type AsyncWorker<P, S, T, U = any> = (params: P, dispatch: Dispatch<T>, getState: () => T, extra: U) => MaybePromise<S>;
                                                                             ~~~~~~~~~~~
```

see https://github.com/reactjs/redux/issues/1342
see https://github.com/reduxjs/redux-thunk/issues/169

**Comment by xdave** [comment]
Hmm, I'm not currently requiring redux >=4 as a peer dependency, but perhaps I should.

**Comment by septs** [comment]
I want to upgrade to `redux@4.0.0`

**Comment by xdave** [comment]
@septs Working on it.

**Comment by xdave** [comment]
@septs  The typings for `redux-thunk` have not yet been updated to support `redux@4.0.0`.  They're working on it, and I'm just waiting for the release.  When it's updated, I'll fix.

**Comment by xdave** [comment]
OK looks like redux-thunk 2.3.0 was released.  I'll do some testing tonight.

**Comment by xdave** [comment]
@septs I've published `2.0.0-beta.10`, please  try it out: `npm i typescript-fsa-redux-thunk@beta`.

FYI: See [src/examples.ts](https://github.com/xdave/typescript-fsa-redux-thunk/blob/v2/src/examples.ts#L95) for how to get `store.dispatch()` to work, as there were some breaking changes with redux-thunk@2.3.x.

**Comment by jussikinnula** [comment]
OK, I guess we could close the issue as "WON'T FIX". We are using workaround anyway at the moment, and we'll upgrade to v2 as soon as we can (in terms of "product development"). Thanks anyway! :-)
